daemon/cluster/services.go

```diff
@@ -134,6 +138,35 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
+			switch serviceSpec.Task.GetGeneric().Kind {
+			case string(types.RuntimePlugin):
+				if s.TaskTemplate.PluginSpec == nil {
+					return errors.New("plugin spec must be set")
+				}
+				p := runtime.GenericPluginPayload{
+					Spec: *s.TaskTemplate.PluginSpec,
+				}
+				if encodedAuth != "" {
+					p.PullOptions = runtime.PluginPullOptions{RegistryAuth: encodedAuth}
```

Comment:
> IIRC the redaction is implicit because the JSON API version of the structure does not contain RegistryAuth

I'm not sure I completely understand. There is no JSON version of the structure containing the registry auth.

But in any case, if we remove this, should we still wrap `PluginSpec` in another struct?